Adding nullable field to vars.graphql InputType breaks tests

I am new to graphql but I think adding a nullable embedded input type should not break the tests since it is nullable and not required but instead it is checking for the required fields inside the nullable input type.

```
// vars.graphql

type Query {
  optionalIntArg(i: Int): Boolean!
  intArg(i: Int!): Boolean!
  stringArg(i: String): Boolean!
  boolArg(i: Boolean!): Boolean!
  floatArg(i: Float!): Boolean!
  idArg(i: ID!): Boolean!
  scalarArg(i: Custom!): Boolean!
  structArg(i: InputType!): Boolean!
  defaultStructArg(i: InputType! = { name: "foo" }): Boolean!
  arrayArg(i: [InputType!]): Boolean!
}

input InputType {
  embedded: Embedded
  name: String!
  nullName: String
}

input Embedded {
  name: String!
}

scalar Custom
```

```
--- FAIL: TestValidateVars (0.00s)
    --- FAIL: TestValidateVars/input_object (0.00s)
        --- FAIL: TestValidateVars/input_object/null_required_field (0.00s)
            /Users/cbelsole/go/src/github.com/vektah/gqlparser/validator/vars_test.go:103: 
                	Error Trace:	vars_test.go:103
                	Error:      	Error message not equal:
                	            	expected: "input: variable.var.name cannot be null"
                	            	actual  : "input: variable.var.embedded.name cannot be null"
                	Test:       	TestValidateVars/input_object/null_required_field
```

--- a/validator/testdata/vars.graphql
+++ b/validator/testdata/vars.graphql
@@ -14,6 +14,11 @@ type Query {
 input InputType {
     name: String!
     nullName: String
+    nullEmbedded: Embedded
+}
+
+input Embedded {
+    name: String!
 }
 
 scalar Custom

--- a/validator/vars_test.go
+++ b/validator/vars_test.go
@@ -79,12 +79,12 @@ func TestValidateVars(t *testing.T) {
 			q := gqlparser.MustLoadQuery(schema, `query foo($var: InputType!) { structArg(i: $var) }`)
 			vars, gerr := validator.VariableValues(schema, q.Operations.ForName(""), map[string]interface{}{
 				"var": map[string]interface{}{
-					"name": "foobar",
-					"nullName":nil,
+					"name":     "foobar",
+					"nullName": nil,
 				},
 			})
 			require.Nil(t, gerr)
-			require.EqualValues(t, map[string]interface{}{"name": "foobar","nullName":nil}, vars["var"])
+			require.EqualValues(t, map[string]interface{}{"name": "foobar", "nullName": nil}, vars["var"])
 		})
 
 		t.Run("missing required values", func(t *testing.T) {
@@ -103,6 +103,17 @@ func TestValidateVars(t *testing.T) {
 				},
 			})
 			require.EqualError(t, gerr, "input: variable.var.name cannot be null")
+		})
+
+		t.Run("null embedded input object", func(t *testing.T) {
+			q := gqlparser.MustLoadQuery(schema, `query foo($var: InputType!) { structArg(i: $var) }`)
+			_, gerr := validator.VariableValues(schema, q.Operations.ForName(""), map[string]interface{}{
+				"var": map[string]interface{}{
+					"name":         "foo",
+					"nullEmbedded": nil,
+				},
+			})
+			require.Nil(t, gerr)
 		})
 
 		t.Run("unknown field", func(t *testing.T) {

